contracts/contracts/token/OUSD.sol

```diff
@@ -584,20 +609,111 @@ contract OUSD is Initializable, InitializableERC20Detailed, Governable {
             ? MAX_SUPPLY
             : _newTotalSupply;
 
-        _rebasingCreditsPerToken = _rebasingCredits.divPrecisely(
-            _totalSupply.sub(nonRebasingSupply)
-        );
+        _rebasingCreditsPerToken =
+            (_rebasingCredits * 1e18) /
+            (_totalSupply - nonRebasingSupply);
 
         require(_rebasingCreditsPerToken > 0, "Invalid change in supply");
 
-        _totalSupply = _rebasingCredits
-            .divPrecisely(_rebasingCreditsPerToken)
-            .add(nonRebasingSupply);
+        _totalSupply =
+            ((_rebasingCredits * 1e18) / _rebasingCreditsPerToken) +
+            nonRebasingSupply;
 
         emit TotalSupplyUpdatedHighres(
             _totalSupply,
             _rebasingCredits,
             _rebasingCreditsPerToken
         );
     }
+
+    function delegateYield(address from, address to)
+        external
+        onlyGovernor
+    {
+        require(from != to, "Cannot delegate to self");
+        require(
+            yieldFrom[to] == address(0) &&
+                yieldTo[to] == address(0) &&
+                yieldFrom[from] == address(0) &&
+                yieldTo[from] == address(0),
+            "Blocked by existing yield delegation"
+        );
+        RebaseOptions stateFrom = rebaseState[from];
+        RebaseOptions stateTo = rebaseState[to];
+
+        require(
+            stateFrom == RebaseOptions.NotSet ||
+                stateFrom == RebaseOptions.StdNonRebasing ||
+                stateFrom == RebaseOptions.StdRebasing,
+            "Invalid rebaseState from"
+        );
+
+        require(
+            stateTo == RebaseOptions.NotSet ||
+                stateTo == RebaseOptions.StdNonRebasing ||
+                stateTo == RebaseOptions.StdRebasing,
+            "Invalid rebaseState to"
+        );
+
+
+        if (
+            alternativeCreditsPerToken[from] == 0 &&
+            (stateFrom == RebaseOptions.NotSet ||
+                stateFrom == RebaseOptions.StdRebasing)
+        ) {
+            _rebaseOptOut(from);
+        }
+        if (
+            alternativeCreditsPerToken[to] > 0 &&
+            (stateTo == RebaseOptions.NotSet ||
```

Comment:
We should remove these state checks and just look at the `alternativeCreditsPerToken`

Reply:
Agreed these are redundant: https://github.com/OriginProtocol/origin-dollar/pull/2298/commits/31c80785df784044ed9d8105018e3dd24ddbb892